lib/config/migration.ts

```diff
@@ -104,6 +104,31 @@ export function migrateConfig(config: RenovateConfig): MigratedConfig {
         }
       }
     }
+    const languages = [
+      'docker',
+      'dotnet',
+      'golang',
+      'java',
+      'js',
+      'node',
+      'php',
+      'python',
+      'ruby',
+      'rust',
+    ];
+    for (const language of languages) {
+      if (is.nonEmptyObject(migratedConfig[language])) {
+        logger.debug(`Migrating ${language} config`);
+        migratedConfig.packageRules ??= [];
+        const currentContent = migratedConfig[language] as any;
+        const packageRule = {
+          matchCategories: [language],
+          ...currentContent,
+        };
+        migratedConfig.packageRules.push(packageRule);
```

Comment:
```suggestion
        migratedConfig.packageRules.unshift(packageRule);
```
because they where applied first previously? Otherwise they would now override other package rules?